test/common/grpc/codec_test.cc

```diff
@@ -106,6 +107,41 @@ TEST(GrpcCodecTest, decodeInvalidFrame) {
   EXPECT_EQ(size, buffer.length());
 }
 
+// A frame of nothing followed by a frame of 0's followed then invalid
+TEST(GrpcCodecTest, decodeNullByteFrameWithInvalidFrameAfterward) {
+  const std::string data("\000\000\000\000\0000000", 9);
+  Buffer::OwnedImpl buffer(data.data(), data.size());
+  std::vector<Frame> frames;
+  Decoder decoder;
+  ENVOY_LOG_MISC(trace, "Buffer: {}", buffer.toString());
+  EXPECT_FALSE(decoder.decode(buffer, frames));
+  // When the decoder doesn't successfully decode, it puts valid frames up until
+  // an invalid frame into output frame vector.
+  EXPECT_NE(0, frames.size());
+}
+
+// If there is valid frames followed by an invalid frame, the decoder will succesfully put the valid
+// frame in the output and return false due to the invalid frame
+TEST(GrpcCodecTest, decodeValidFramesWithInvalidFrameAfterward) {
```

Comment:
nit: can you give a succinct name for readability?
DecodeMultipleFramesInvalid maybe? (Decoding multiple valid frames is covered below)

Reply:
Yeah, I think the key for this test is letting code owner and developers know about this null byte frame functionality, and if it is supposed to actually decode null bytes into a frame. I'll reword and be verbose here.

Reply:
I'm gonna add a todo here asking if this test should really pass.

Reply:
I think this should be called "DecodeValidFrameWithInvalidFrameAfterward" and other unit test with null bytes and a bunch of zeros should be called "DecodeMultipleFramesInvalid"